pkg/apis/core/validation/shoot.go

```diff
@@ -1541,8 +1541,13 @@ func ValidateKubeControllerManager(kcm *core.KubeControllerManagerConfig, networ
 			}
 		}
 
-		if podEvictionTimeout := kcm.PodEvictionTimeout; podEvictionTimeout != nil && podEvictionTimeout.Duration <= 0 {
-			allErrs = append(allErrs, field.Invalid(fldPath.Child("podEvictionTimeout"), podEvictionTimeout.Duration, "podEvictionTimeout must be larger than 0"))
+		// TODO(plkokanov): Remove this check after support for Kubernetes 1.32 is dropped.
+		if podEvictionTimeout := kcm.PodEvictionTimeout; podEvictionTimeout != nil {
+			if k8sGreaterEqual133, _ := versionutils.CheckVersionMeetsConstraint(version, ">= 1.33"); k8sGreaterEqual133 {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("podEvictionTimeout"), kcm.PodEvictionTimeout, "podEvictionTimeout is no longer supported by Gardener starting from Kubernetes 1.33"))
```

Comment:
If we introduce `podEvictionTimeout := kcm.PodEvictionTimeout` lets use it in the if block for concistency with line 1548 and 1549
```suggestion
				allErrs = append(allErrs, field.Invalid(fldPath.Child("podEvictionTimeout"), podEvictionTimeout, "podEvictionTimeout is no longer supported by Gardener starting from Kubernetes 1.33"))
```